app/ordered_map.go

```diff
@@ -82,6 +82,27 @@ func (om *OrderedMap[K, V]) MustGet(key K) V {
 	return value
 }
 
+func (om *OrderedMap[K, V]) GetOrSetDefault(key K, defaultValue V) (V, bool) {
+	if value, exists := om.Get(key); exists {
+		return value, false
+	} else {
+		om.Set(key, defaultValue)
+		return om.MustGet(key), true
+	}
+}
+
+// Iterate returns a channel that yields key-value pairs in insertion order
+func (om *OrderedMap[K, V]) Iterate() <-chan Pair[K, V] {
+	ch := make(chan Pair[K, V])
+	go func() {
+		for _, key := range om.keys {
+			ch <- Pair[K, V]{Key: key, Value: om.MustGet(key)}
+		}
+		close(ch)
+	}()
+	return ch
+}
```

Comment:
Keep in mind that this solution does a lot of shuffling data through the tread safe queueue (most probably using mutex) - internally the channel is implemented effectivelly as a thread safe queue.

This involves literally copying **value** of every single element of a collection the iteration goes through in to the channel (in this case copying every single `Pair[K, V]` value) over to channel queue ...

Looking at this solely from the performance perspective, it is not worth it.
Though I do understand that given the resulting performance impact in our use case, it is practivally irrelevant + this approach makes code more lean and so more readable ...